protocol/protocol.tex

```diff
@@ -4299,30 +4744,169 @@
+
+\vspace{1ex}
+$\AuthSignPublic \typecolon \GroupP$, the \incomingViewingKey
+$\InViewingKey \typecolon \InViewingKeyType{Orchard}$, and the \outgoingViewingKey
+$\OutViewingKey \typecolon \OutViewingKeyType{Orchard}$ are then derived as:
+
+\vspace{-0.5ex}
+\begin{tabular}{@{\hskip 1.7em}r@{\;}l}
+  $\AuthSignPublic$ &$:= \SpendAuthSigDerivePublic{Orchard}(\AuthSignPrivate)$ \\
+  $\InViewingKey$ &$:= \CommitIvk{\CommitIvkRand}\big(\reprP\Of{\ExtractP{\AuthSignPublic}}, \reprP\Of{\NullifierKey}\kern-0.08em\big)$ \\
+  $\OutViewingKey$ &$:= \CRHovk\big(\CommitIvkRand, \reprP\Of{\ExtractP{\AuthSignPublic}}, \reprP\Of{\NullifierKey}\kern-0.08em\big)$.
+\end{tabular}
+
+As explained in \crossref{addressesandkeys}, \Orchard allows the efficient
+creation of multiple \diversifiedPaymentAddresses with the same spending
+authority. A group of such addresses shares the same \fullViewingKey,
+\incomingViewingKey, and \outgoingViewingKey.
+
+\introlist
+To create a new \diversifiedPaymentAddress given an \incomingViewingKey
+$\InViewingKey$, repeatedly pick a \defining{\diversifier} $\Diversifier$ uniformly at
+random from $\DiversifierType$ until the \defining{\diversifiedBase}
+$\DiversifiedTransmitBase = \DiversifyHash{Sapling}(\Diversifier)$ is not $\bot$.
+Then calculate the \defining{\diversifiedTransmissionKey} $\DiversifiedTransmitPublic$:
+
+\begin{formulae}
+  \item $\DiversifiedTransmitPublic := \KADerivePublic{Sapling}(\InViewingKey, \DiversifiedTransmitBase)$.
+\end{formulae}
+
+\vspace{-1ex}
+The resulting \diversifiedPaymentAddress is
+$(\Diversifier \typecolon \DiversifierType, \DiversifiedTransmitPublic \typecolon \KAPublicPrimeSubgroup{Sapling})$.
+
+\vspace{1ex}
+For each \spendingKey, there is also a \defining{\defaultDiversifiedPaymentAddress}
+with a ``random-looking'' \diversifier. This allows an implementation that does not expose
+diversified addresses as a user-visible feature, to use a default address that
+cannot be distinguished (without knowledge of the \spendingKey) from one with a random
+\diversifier as above.
+
+\introlist
+Let $\first \typecolon (\byte \rightarrow \maybe{T}) \rightarrow \maybe{T}$
+be as defined in \crossref{concretegrouphashjubjub}. Define:
+\vspace{-0.5ex}
+\begin{formulae}
+  \item $\CheckDiversifier(\Diversifier \typecolon \DiversifierType) := \begin{cases}
+          \bot,         &\caseif \DiversifyHash{Sapling}(\Diversifier) = \bot \\
+          \Diversifier, &\caseotherwise
+        \end{cases}$
+  \item $\DefaultDiversifier(\sk \typecolon \SpendingKeyType) :=
+          \first\big(\fun{i \typecolon \byte}{\CheckDiversifier(\truncate{(\DiversifierLength/8)}(\PRFexpand{\sk}([3, i])))
+                                              \typecolon \maybe{\SubgroupJstar}}\big)$.
+\end{formulae}
+
+For a random \spendingKey, $\DefaultDiversifier$ returns $\bot$ with probability approximately $2^{-256}$;
+if this happens, discard the key and repeat with a different $\SpendingKey$.
```

Comment:
This is copied from Sapling, and needs to be updated for Orchard (`\DiversifyHash{Orchard}` is infallible).

Reply:
I've instead added an Orchard diversifier section to ZIP 32 in #448.